### DISCUSSION TITLE
Can Auth Server (IdentityServer or OpenIddict) be completely separated from Identity Service in eShopOnAbp?

### DISCUSSION THREAD

**Discussion by jflaga** [abpframework/abp#18326] [Q&A] [answered]
In [version 1.0 of eShopOnAbp (Microservice Demo), the Auth Server uses IdentityServer.](https://github.com/abpframework/eShopOnAbp/blob/1.0/apps/auth-server/src/EShopOnAbp.AuthServer/EShopOnAbp.AuthServer.csproj#L17C8-L17C91)

Auth Server is also [dependent on `EShopOnAbp.IdentityService.EntityFrameworkCore` module](https://github.com/abpframework/eShopOnAbp/blob/1.0/apps/auth-server/src/EShopOnAbp.AuthServer/EShopOnAbp.AuthServer.csproj#L27C7-L27C172) from the Identity Service.

![image](https://github.com/abpframework/abp/assets/108422823/ad8e3a04-85c8-403b-8bd7-dce17e1dd2fb)

Is this dependency on Identity Service is unavoidable?

Is there a way to completely separate Auth Server and Identity Service in ABP?

Thanks

**Comment by maliming** [comment]
hi

The authserver has to depend on the openiddict or identityserver4.

**Reply by jflaga** [reply]
Does it need also need to depend on `Volo.Abp.Identity.EntityFrameworkCore`?

**Reply by maliming** [reply, marked as answer]
The user/authentication system needs the `Volo.Abp.Identity.EntityFrameworkCore`.

**Reply by jflaga** [reply]
Okay.. thank you so much @maliming
